proposals/XXXX-atf-selenium-support_review.md

```diff
@@ -0,0 +1,236 @@
+
+```json
+{
+  "SDL.ActivateApp" : {
+    "SendRequest": {
+      "action": [{
+        "id": "#appListItem1", //user by Document.getElementById()
+        "function": "click" // action to be taken by the webdriver
+      }] 
+    }
+  }
+}
+```
+
+
+This example shows how an `ExpectResponse` behavior would be overwritten for the activate app response. After the response is received the WebDriver will validate that the app is active via the displayed app name.
+
+```json
+{
+  "SDL.ActivateApp" : {
+    "ExpectResponse": {
+      "validate": [{
+        "id": "#appListHeader", //user by Document.getElementById()
+        "function": <App Name> // text to validate the contents of the field
+      }] 
+    }
+  }
+}
+```
+
+### Scope Of Test Scripts
+
+ATF has a very large collection of tests that are implemented with varying design patterns. It is hard to gauge the scope of how much work it would take to make every ATF test compatible with the Selenium WebDriver. The author suggests that the initial implementation of this proposal covers basic smoke test cases such as registering an app, activating an app, displaying some data, and performing some sort of button press type interactions.
+
+### Architecture Diagram
+
+![Architecture Diagram](../assets/proposals/0340-atf-selenium-support/ATFSelenium.png)
+
+## Potential downsides
+
+This proposal includes a lot of components that have to interact and communicate with one another. Maintaining and developing this proposal would take a lot of work but the author thinks it would be extremely beneficial in helping automate our IVI related components tests. The less time we spend manually testing features, the more time we have for bug fixes and feature development.
+
+## Impact on existing code
+
+### ATF
+
+- ATF will need to implement some new websocket servers/clients to communicate with the different components seen in the architecture diagram.
+- ATF request/response methods should be extended to handle the RPC switching feature for this proposal.
+- ATF will need to handle the lifecycle of starting and stopping the python Selenium application.
+
+### ATF Test Scripts
+- Some scripts may need to be modified more than others to accommodate the Selenium testing feature.
+
+### Python Selenium Application
+- Websockets should be set up to communicate with the ATF test scripts. The WebDriver will read messages that contain the actions defined in the `RPC Switching` section.
+
+### Generic HMI
+- Generic HMI might need to be updated to include the id property for applicable html elements to allow for the Selenium WebDriver to easily find these elements that need validation or interaction.
+
+
+## Alternatives considered
+
+An alternative solution would be to first create automated Selenium tests that only interact with the Generic HMI. This simplified solution wouldn't involve SDL Core or ATF.
```

Comment:
An another alternative is  "HMI testing application" approach:
 - HMI testing application is specific application which can perform actions and verifications on specific HMI implementation
 - ATF can connect and conduct of test run on specific HMI testing application via web socket connection
 - ATF can perform testing using emulated HMI testing application (current approach)
 - ATF test scripts are independent from details of HMI implementation